### ISSUE TITLE
Kernel Panic - kext com.driver.apple.usb.AppleUSBHostPlatformProperties declares no com.apple.kpi.* dependencies.

### ISSUE THREAD

**Issue by Arrnav** [tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579#1]
![KP](https://user-images.githubusercontent.com/23283363/78231335-4497e600-74f0-11ea-9c60-dad6bf11b119.png)

Getting a Kernel Panic using the entire OC Folder provided here. (Only change I did was change ShowPicker=true in config.plist.

Config -
Dell G3 3579
Full size HDMI port (Not Type-C)
Intel Core i5-8300H @ 2.30Ghz
12GB DDR4 2667Mhz (8 + 4)
Samsung EVO 970 Pro 512GB NVME SSD (Windows 10 + Existing MacOS 10.15.4 Previously Working fully with Clover)
Samsung 850 EVO 128GB SATA SSD (Storage)
Intel UHD Graphics 630 + NVIDIA GeForce GTX 1050 4GB
Realtek ALC236
Realtek RTL8111
Intel Wireless-AC 9462

EFI (With OC folder) on USB -
[EFI.zip](https://github.com/tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579/files/4420092/EFI.zip)

**Comment by tonyleelyy** [comment]
I’m sorry that my USB ports is customized by myself and those kexts may not be suitable for your computer.
Please try [release v1.0](https://github.com/tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579/releases/tag/v1.0) and see if it can work.
I will fixed that problem in the future version. Thank you!

**Comment by Arrnav** [comment]
> I’m sorry that my USB ports is customized by myself and those kexts may not be suitable for your computer.
> Please try [release v1.0](https://github.com/tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579/releases/tag/v1.0) and see if it can work.
> I will fixed that problem in the future version. Thank you!

Hi, thanks for your prompt reply,

The exactly same kernel panic and issue persists even if I'm using version 1.0

Is there any implementation guide for our USB so I can see what this error is about myself? I can add a commit here as a fix if I can fix this issue myself.

Also, what is the com.driver.apple.usb.AppleUSBHostPlatformProperties? I can't seem to find this kext anywhere, nor is there any information about this kext online.

**Comment by tonyleelyy** [comment]
Hi, I'm wondering if you can boot up and use USB devices with my EFI? Or can you use USB with your clover EFI? 
Cuz I tried to see my boot information but it moves too fast and seems to work perfectly. Maybe it is just a warning without any issue?
Neither can I find the information about _com.driver.apple.usb.AppleUSBHostPlatformProperties_ , but I'm sure it's a kext from apple and it's in /System/Library/Extensions or /Library/Extensions.

**Comment by Arrnav** [comment]
> Hi, I'm wondering if you can boot up and use USB devices with my EFI? Or can you use USB with your clover EFI?
> Cuz I tried to see my boot information but it moves too fast and seems to work perfectly. Maybe it is just a warning without any issue?
> Neither can I find the information about _com.driver.apple.usb.AppleUSBHostPlatformProperties_ , but I'm sure it's a kext from apple and it's in /System/Library/Extensions or /Library/Extensions.

I can boot up fine using Clover and yes USB works perfectly then. 

Here is my fully working Clover folder -

[CLOVER.zip](https://github.com/tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579/files/4425252/CLOVER.zip)

The issue is definitely not just a warning and causes the system to fail to boot with the mentioned kernel panic.

One thing to note is that **in my successful implementation of Clover** I'm using USBInjectAll kext instead of USBPower kext **in your OC**. Maybe I can try to re-implement everything from scratch and add to fixes here if they work well for our laptop? 

Is it that hard for the 3579?

**Comment by tonyleelyy** [comment]
> > Hi, I'm wondering if you can boot up and use USB devices with my EFI? Or can you use USB with your clover EFI?
> > Cuz I tried to see my boot information but it moves too fast and seems to work perfectly. Maybe it is just a warning without any issue?
> > Neither can I find the information about _com.driver.apple.usb.AppleUSBHostPlatformProperties_ , but I'm sure it's a kext from apple and it's in /System/Library/Extensions or /Library/Extensions.
> 
> I can boot up fine using Clover and yes USB works perfectly then.
> 
> Here is my fully working Clover folder -
> 
> [CLOVER.zip](https://github.com/tonyleelyy/OpenCore-Hackintosh-Dell-G3-3579/files/4425252/CLOVER.zip)
> 
> The issue is definitely not just a warning and causes the system to fail to boot with the mentioned kernel panic.
> 
> One thing to note is that **in my successful implementation of Clover** I'm using USBInjectAll kext instead of USBPower kext **in your OC**. Maybe I can try to re-implement everything from scratch and add to fixes here if they work well for our laptop?
> 
> Is it that hard for the 3579?

YES, please add to fixes here and I think anyone who has the same problem will be appreciated.

As for the re-implement, it is totally not hard for the 3579. Maybe you can start with simplifing my config.plist, /kexts and /ACPI folder.

By the way, my EFI is also edited from a previous Dell G3 3579 OC project from GitHub→[link](https://github.com/CerteKim/Dell-G3-3579-Hackintosh-OpenCore). It is made by my friend CerteKim. Please try his version and check if it works so that I can make further troubleshooting.

Thanks again for your contribution to this project.

**Comment by Arrnav** [comment]
SOLVED! Apparently when switching from Clover to Opencore, MacOS /L/E/ has to be in a vanilla state for things to work smoothly and avoid strange errors.

I had Lilu, WG and USBInjectAll (This one caused all the headache I suppose) in my /L/E/ from my previous working clover installed MacOS.

First I removed all third party kexts from /L/E/ using an APFS disk access tool in Windows, then I rebooted using previous Clover on EFI into Recovery, rebuilt cache using kextcache -i command and finally rebooted into MacOS from OC, no more error, smoothly booted into the system!
